services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/AbstractMappingAndDelegatingCommandHandler.java

```diff
@@ -103,8 +114,21 @@ public AbstractMappingAndDelegatingCommandHandler(
                             commandContext.getTracingSpan());
                     return tenantObject;
                 })
-                .compose(tenantObject -> commandTargetMapper.getTargetGatewayAndAdapterInstance(command.getTenant(),
-                        command.getDeviceId(), commandContext.getTracingContext()))
+                .compose(tenantObject -> {
+                    // check whether the handler messaging type is equal to the messaging type of the tenant (if set)
+                    final MessagingType tenantMessagingType = Optional
+                            .ofNullable(tenantObject.getProperty(TenantConstants.FIELD_EXT, JsonObject.class))
+                            .map(ext -> ext.getString(TenantConstants.FIELD_EXT_MESSAGING_TYPE))
+                            .map(MessagingType::valueOf).orElse(null);
+                    if (tenantMessagingType != null && getMessagingType() != tenantMessagingType) {
+                        log.info("command received via {} but tenant is configured to use {} [{}]", getMessagingType(),
+                                tenantMessagingType, commandContext.getCommand());
+                        commandContext.getTracingSpan().log(String.format("command received via %s but tenant is configured to use %s",
```

Comment:
FMPOV we should log these two values to the span in any case at the beginning of the method. During migration this will probably become handy to track down routing problems ...

Reply:
I've extended the span log message created in `Command#logtoSpan` ("received command message via AMQP/Kafka").
The tenant default messaging type is only available after having fetched the tenant object, so I would keep it here. Whether this gets logged a few milliseconds earlier or later doesn't matter much anyway, I think.